editors/emacs/lambdapi-mode.el

```diff
@@ -99,11 +99,102 @@
           (setq lambdapi-current-line-number new-line-number)
           (run-hooks 'changed-line-hook)))))
 
-(defun lambdapi-create-goals-buffer ()
-  (let ((goalsbuf (get-buffer-create "*Goals*"))
-        (goalswindow (split-window nil -10 'below)))
-    (set-window-buffer goalswindow goalsbuf)
-    (set-window-dedicated-p goalswindow 't)))
+
+(defgroup lambdapi nil
+  "LambdaPi is proof assistant based on λΠ-calculus modulo rewriting"
+  :group 'languages)
+
+(defcustom lambdapi-goals-window-side 'right
+  "Side at which to show goals window"
+  :type '(choice (const :tag "right"  right)
+                 (const :tag "left"   left)
+                 (const :tag "top"    top)
+                 (const :tag "bottom" bottom))
+  :group 'lambdapi)
+
+(defcustom lambdapi-goals-window-height-ratio 0.30
+  "Ratio of height taken by Goals window if split top or bottom"
+  :type '(float)
+  :group 'lambdapi)
+
+(defcustom lambdapi-goals-window-width-ratio 0.5
+  "Ratio of width taken by Goals window if split left or right"
+  :type '(float)
+  :group 'lambdapi)
+
+(defcustom lambdapi-goals-window-min-width 40
+  "Minimum width of Goals window"
+  :type '(integer)
+  :group 'lambdapi)
+
+(defcustom lambdapi-goals-window-min-height 4
+  "Minimum height of Goals window"
+  :type '(integer)
+  :group 'lambdapi)
```

Comment:
I think these `defcustom` variables could be defined in `lambdapi-vars.el`, rather than here. But you seem to be more acquainted with elisp than I am, so you might have better reasons to keep them here.

Reply:
```
;;; Commentary:
;;
;; The lists defined are mainly used for syntax colouring and completion.
```
I didn't put it there, because the above comment said that variables defined there are only for syntax and completion. Moreover, variables are only used in `lambdpi-refresh-window-layout` function. So I thought of defining them near it.

Reply:
OK. Sounds reasonable, I'm not fixed on the organisation of elisp files anyway. I just wonder how someone learns the existence of modifiable variables (`defcustom`); we may have to document them in the user manual.

Reply:
Yes, thanks to add this in the user manual.